spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/credentials/AwsCodeCommitCredentialProvider.java

```diff
@@ -0,0 +1,374 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.config.server.credentials;
+
+import static org.springframework.util.StringUtils.hasText;
+
+import java.net.URI;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.eclipse.jgit.errors.UnsupportedCredentialItem;
+import org.eclipse.jgit.transport.CredentialItem;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.URIish;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.util.ValidationUtils;
+
+/**
+ * Provides a jgit {@link CredentialsProvider} implementation that can provide
+ * the appropriate credentials to connect to an AWS CodeCommit repository.
+ * <p>
+ * From the command line, you can configure git to use AWS code commit with a
+ * credential helper. However, jgit does not support credential helper commands,
+ * but it does provider a CredentialsProvider abstract class we can extend.
+ * </p>
+ * Connecting to an AWS CodeCommit (codecommit) repository requires an AWS access key
+ * and secret key. These are used to calculate a signature for the git request. The
+ * AWS access key is used as the codecommit username, and the calculated signature
+ * is used as the password. The process for calculating this signature is documented
+ * very well at http://docs.aws.amazon.com/general/latest/gr/signature-version-4.html.
+ * </p>
+ * 
+ * @author Don Laidlaw
+ *
+ */
+public class AwsCodeCommitCredentialProvider extends CredentialsProvider {
+
+	private static final String SHA_256 = "SHA-256";		//$NON-NLS-1$
+	private static final String UTF8 = "UTF8";				//$NON-NLS-1$
+	private static final String HMAC_SHA256 = "HmacSHA256";	//$NON-NLS-1$
+	private static final char[] hexArray = "0123456789abcdef".toCharArray();	//$NON-NLS-1$
+	
+	protected Log logger = LogFactory.getLog(getClass());
+
+	/**
+	 * The AWSCredentialsProvider will be used to provide the access key and secret
+	 * key if they are not specified.
+	 */
+	private AWSCredentialsProvider awsCredentialProvider;
+	
+	/**
+	 * If the access and secret keys are provided, then the AWSCredentialsProvider will 
+	 * not be used. The username is the awsAccessKeyId.
+	 */
+    private String username;
```

Comment:
Completed. As well as all the other locations where spaces were used to indent. Also matched other code style preferences as per Spring Boot Java conventions.